app/models/edition.rb

```diff
@@ -643,6 +644,10 @@ def set_public_timestamp
                             end
   end
 
+  def set_auth_bypass_id
+    self.auth_bypass_id = SecureRandom.uuid
```

Comment:
Just a quick comment as I know you're still working on this, we shouldn't have this set on before_save as it'll mean every edit to the document changes the auth_bypass_id and thus invalidates the token.